### PR TITLE
JSON codec call & sdp state command and response

### PR DESCRIPTION
**Pull request by Roog**
Hi!
Thanks so much for including the other pull request for #973 in the master. I thought i give it a try to include call information in the JSON "dump". 

```
{
   "sip:rogersan@sip.com":{
      "cuser":"rogersan-0x7fdb0944ec50",
      "selected_ua":true,
      "aor":"sip:rogersan@sip.com",
      "display_name":"",
      "settings":{
         "sip_nat_outbound":[

         ],
         "stun_host":"",
         "stun_port":0,
         "stun_user":"rogersan",
         "answer_mode":"manual",
         "call_transfer":true,
         "packet_time":20
      },
      "registration":{
         "id":0,
         "state":true,
         "code":200,
         "srv":"None",
         "ipv":"v4",
         "interval":3664,
         "q_value":0.000000
      },
      "call":{
         "id":"330ec9df0ee769a9",
         "state":"OUTGOING",
         "peer_uri":"sip:peer-dest@sip.com",
         "direction":"outgoing",
         "on_hold":false,
         "start_time":0.000000,
         "request_time":0.000000,
         "local":{
            "tool":"baresip 1.0.0"
         },
         "remote":{

         },
         "media":{
            "audio":{
               "local_formats":{
                  "0":[
                     "PCMU",
                     8000,
                     1,
                     "",
                     true,
                     "0"
                  ],
                  "8":[
                     "PCMA",
                     8000,
                     1,
                     "",
                     true,
                     "8"
                  ],
                  "101":[
                     "telephone-event",
                     8000,
                     1,
                     "0-15",
                     true,
                     "101"
                  ]
               },
               "remote_formats":{

               },
               "local_attributes":{
                  "label":"1",
                  "rtcp-rsize":true,
                  "ssrc":"2141783550 cname:sip:rogersan@sip.com",
                  "minptime":"20",
                  "ptime":"20"
               },
               "remote_attributes":{

               }
            },
            "protocol":"RTP\/AVP"
         },
         "local_media":{

         }
      }
   }
}
```
The method should append the "call" node. 

And this pull-request is dependant on another pull-request for the "re" library [JSON formatted response with sdp info initiated #250](https://github.com/creytiv/re/pull/250).

Please let me know what you think @alfredh . And maybe @premultiply has some comments on the suggested JSON format?

Some questions if you could point me in the right direction:
- can we find out whats the current negotiated algorithm in a session is?
- how to properly identify if an attribute is only consisting of "numbers" as a "char"?

Best regards
Roger

### CODE REVIEW

**Comment by alfredh** [comment]
can you try to rewrite this PR, so that it does not require any changes in libre ?

if you make changes in libre, it is not certain how long time it will take until those
changes are in master. it can easily take many months, maybe a year, maybe never.

File: modules/aac/aac.c [hunk 1]
```diff
@@ -203,58 +203,58 @@ static int module_init(void)
 	aac.crate = aac_samplerate;
 
 	switch (aac_aot) {
-	case AOT_AAC_LC:
```
Comment (premultiply) [inline modules/aac/aac.c:206]: I think these changes are not related to this PR ;)

**Comment by Roog** [comment]
> can you try to rewrite this PR, so that it does not require any changes in libre ?
> 
> if you make changes in libre, it is not certain how long time it will take until those
> changes are in master. it can easily take many months, maybe a year, maybe never.

Hi @alfredh good to hear from you :) I Hope all is well.
I did not know, but yes i will try to rewrite this without libre!

File: modules/aac/aac.c (hunk 1 above)
Comment (Roog) [reply modules/aac/aac.c:206]: That's accurate ;) just came by it and thought I'd adjust the code formatting.. I have some OCD when it comes to no use white spaces hanging around :) thinking the one approving the code will have to say yay or nay. Did you have any thoughts about the Json information and did you need anything specific there?

Comment (premultiply) [reply modules/aac/aac.c:206]: I hope I can have a deeper look inside when I am at the office for a few days next week.
I do not have the equipment and tools at home to test anything related to baresip.

**Comment by juha-h** [comment]
> can you try to rewrite this PR, so that it does not require any changes in libre ?
> 
> if you make changes in libre, it is not certain how long time it will take until those
> changes are in master. it can easily take many months, maybe a year, maybe never.

It is not good if baresip development is held hostage by libre, if libre is not anymore actively maintained. Why can't we fork libre and keep on developing it ourselves according to baresip needs?

File: src/account.c [hunk 2]
```diff
@@ -1216,11 +1216,10 @@ uint16_t account_stun_port(const struct account *acc)
...
 	switch (mode) {
-
-	case ANSWERMODE_MANUAL: return "manual";
-	case ANSWERMODE_EARLY:  return "early";
-	case ANSWERMODE_AUTO:   return "auto";
-	default: return "???";
+		case ANSWERMODE_MANUAL: return "manual";
+		case ANSWERMODE_EARLY:  return "early";
+		case ANSWERMODE_AUTO:   return "auto";
+		default: return "???";
```
Comment (alfredh) [inline src/account.c:1222]: this part changes only the formatting, and is not related to the main PR.

**Comment by alfredh** [comment]
hi @Roog 

your patch contains a lot of unrelated formatting changes.

please create a new PR that only contains the relevant changes.

**Comment by Roog** [comment]
Sure! Sorry about that! 
Should I still try to not reference the "re" library or use the forked one on the baresip GitHub? 
Best regards
Roger

**Comment by alfredh** [comment]
please use this one: https://github.com/baresip/re
